src/client/format.ts

```diff
@@ -8,11 +8,13 @@ import {
 } from "vscode";
 import { execV } from "./exec";
 import { fullDocumentRange, getVConfig } from "./utils";
+import { writeFileSync } from "fs";
 
 function format(document: TextDocument): Promise<TextEdit[]> {
 	return new Promise((resolve, reject) => {
 		const vfmtArgs = getVConfig().get("format.args", "");
 		const args = ["fmt", vfmtArgs, document.fileName];
+		writeFileSync(document.fileName, document.getText())
```

Comment:
👋 I might be missing something but it's best for us (vs code) to write to a temp-file, not the file itself.

Reply:
👋 Hi @jrieken, At first that was what we planned and implemented, but the problem was there :  ☝️ https://github.com/0x9ef/vscode-vlang/pull/128#issuecomment-584165244

Reply:
hm... This can then cause conflicts in VS Code, e.g when a file changes on disk and when there are unsaved changes in the editor, a conflict situation is very likely. Given those constraint, I don't think that this is a good solution anymore. Is there no other way to invoke the formatter? We have seen many cli-tools to which you can pipe the contents of a file (instead of loading it from disk)

Reply:
@jrieken yes, you are right. Saving to the file directly does not work (see my message below).